node/src/types/chainspec/accounts_config.rs

```diff
@@ -95,37 +95,142 @@ impl FromBytes for AccountConfig {
     }
 }
 
+#[derive(PartialEq, Eq, Serialize, Deserialize, DataSize, Debug, Copy, Clone)]
+pub struct DelegatorConfig {
+    validator_public_key: PublicKey,
+    delegator_public_key: PublicKey,
+    balance: Motes,
+    delegated_amount: Motes,
+}
+
+impl DelegatorConfig {
+    pub fn new(
+        validator_public_key: PublicKey,
+        delegator_public_key: PublicKey,
+        balance: Motes,
+        delegated_amount: Motes,
+    ) -> Self {
+        Self {
+            validator_public_key,
+            delegator_public_key,
+            balance,
+            delegated_amount,
+        }
+    }
+
+    pub fn validator_public_key(&self) -> PublicKey {
+        self.validator_public_key
+    }
+
+    pub fn delegator_public_key(&self) -> PublicKey {
+        self.delegator_public_key
+    }
+
+    pub fn balance(&self) -> Motes {
+        self.balance
+    }
+
+    pub fn delegated_amount(&self) -> Motes {
+        self.delegated_amount
+    }
+}
+
+impl Distribution<DelegatorConfig> for Standard {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> DelegatorConfig {
+        let validator_public_key = SecretKey::ed25519(rng.gen()).into();
+        let delegator_public_key = SecretKey::ed25519(rng.gen()).into();
+
+        let mut u512_array = [0u8; 64];
+        rng.fill_bytes(u512_array.as_mut());
+        let balance = Motes::new(U512::from(u512_array));
+
+        rng.fill_bytes(u512_array.as_mut());
+        let delegated_amount = Motes::new(U512::from(u512_array));
+
+        DelegatorConfig::new(
+            validator_public_key,
+            delegator_public_key,
+            balance,
+            delegated_amount,
+        )
+    }
+}
+
+impl ToBytes for DelegatorConfig {
+    fn to_bytes(&self) -> Result<Vec<u8>, bytesrepr::Error> {
+        let mut buffer = bytesrepr::allocate_buffer(self)?;
+        buffer.extend(self.validator_public_key.to_bytes()?);
+        buffer.extend(self.delegator_public_key.to_bytes()?);
+        buffer.extend(self.balance.to_bytes()?);
+        buffer.extend(self.delegated_amount.to_bytes()?);
+        Ok(buffer)
+    }
+
+    fn serialized_length(&self) -> usize {
+        self.validator_public_key.serialized_length()
+            + self.delegator_public_key.serialized_length()
+            + self.balance.serialized_length()
+            + self.delegated_amount.serialized_length()
+    }
+}
+
+impl FromBytes for DelegatorConfig {
```

Comment:
It'd be worthwhile adding a bytesrepr roundtrip test for this (and it seems like `AccountsConfig` could use one too).

Reply:
I did it at the outer layer (AccountsConfig) which should cover the sub graphs as well.